Dedupe no parsing __torch_function__ handler

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75161
* #75154
* __->__ #75159
* #75142
* #75141

Now there is truly only one way to call __torch_function__
and that is via handle_torch_function_no_python_arg_parser

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
Dedupe no parsing __torch_function__ handler

Now there is truly only one way to call __torch_function__
and that is via handle_torch_function_no_python_arg_parser

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Dedupe no parsing __torch_function__ handler"


Now there is truly only one way to call __torch_function__
and that is via handle_torch_function_no_python_arg_parser

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Dedupe no parsing __torch_function__ handler"


Now there is truly only one way to call __torch_function__
and that is via handle_torch_function_no_python_arg_parser

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

peterbell10 commented on torch/csrc/utils/python_arg_parser.cpp:
```diff
@@ -184,24 +184,11 @@ auto combine_self_args(PyObject *self, PyObject *args) -> py::tuple {
 }
 
 auto handle_torch_function(PyObject* self, const std::string& func_name, PyObject* args, PyObject* kwargs, PyObject* torch_api, const std::string& module_name) -> PyObject* {
+  std::vector<py::handle> overloaded_args{self};
```
NIT: Consider changing `handle_torch_function_no_python_arg_parser` to take `ArrayRef` for `overloaded_args` so we don't need the extra allocation here.

ezyang committed:
Update on "Dedupe no parsing __torch_function__ handler"


Now there is truly only one way to call __torch_function__
and that is via handle_torch_function_no_python_arg_parser

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge this
